src/java/htsjdk/samtools/SAMLineParser.java

```diff
@@ -170,6 +177,21 @@ private int parseInt(final String s, final String fieldName) {
         }
         return ret;
     }
+    
+    private int parseFlag(final String s, final String fieldName) {
+        final int ret;
+        final SamFlagField samFlagField = this.samFlagField.orElse(SamFlagField.getSamFlagField(s));
+        try {
+            ret = samFlagField.parse(s);
```

Comment:
I'd find it easier to read like this (ie eliminate the useless temp declarations outside of the `try` block)
```
try{
   final SamFlagField samFlagField = this.samFlagField.orElse(SamFlagField.getSamFlagField(s));
   return samFlagField.parse(s);
} catch ...
```